lib/ansible/plugins/strategy/free.py

```diff
@@ -226,8 +226,10 @@ def run(self, iterator, play_context):
                             new_blocks = self._load_included_file(included_file, iterator=iterator)
                     except AnsibleError as e:
                         for host in included_file._hosts:
-                            iterator.mark_host_failed(host)
-                        display.warning(to_text(e))
+                            if host.name not in self._tqm._failed_hosts:
+                                self._tqm._failed_hosts[host.name] = True
+                                iterator.mark_host_failed(host)
+                        display.error(to_text(e), wrap_text=False)
```

Comment:
This makes the failures handling consistent with how `linear` strategy does it. I am not sure why that was different or if that was intentional.